src/nixpacks/builder/docker/utils.rs

```diff
@@ -24,28 +24,44 @@ pub fn get_cache_mount(
     }
 }
 
-pub fn get_copy_command(files: &[String], app_dir: &str) -> String {
+pub fn get_copy_commands(files: &[String], app_dir: &str) -> Vec<String> {
     if files.is_empty() {
-        String::new()
+        Vec::with_capacity(0)
     } else {
-        format!("COPY {} {}", files.join(" "), app_dir)
+        files
+            .iter()
+            .map(|file| {
+                let mut file_in_app_dir = app_dir.to_string();
+                if !file_in_app_dir.ends_with('/') {
+                    file_in_app_dir.push('/');
+                }
+                file_in_app_dir.push_str(file.strip_prefix("./").unwrap_or(file));
+                format!("COPY {file} {file_in_app_dir}")
+            })
+            .collect()
     }
 }
 
-pub fn get_copy_from_command(from: &str, files: &[String], app_dir: &str) -> String {
+pub fn get_copy_from_commands(from: &str, files: &[String], app_dir: &str) -> Vec<String> {
     if files.is_empty() {
-        format!("COPY --from=0 {} {}", app_dir, app_dir)
+        vec![format!("COPY --from=0 {} {}", app_dir, app_dir)]
     } else {
-        format!(
-            "COPY --from={} {} {}",
-            from,
-            files
-                .iter()
-                .map(|f| f.replace("./", app_dir))
-                .collect::<Vec<_>>()
-                .join(" "),
-            app_dir
-        )
+        files
+            .iter()
+            .map(|file| {
+                let mut file_in_app_dir = app_dir.to_string();
+                if !file_in_app_dir.ends_with('/') {
+                    file_in_app_dir.push('/');
+                }
+                file_in_app_dir.push_str(file.strip_prefix("./").unwrap_or(file));
+                let file = if file.starts_with("./") {
+                    &file_in_app_dir
+                } else {
+                    file
+                };
```

Comment:
Nit: This can be cleaned up a bit by using `Path::join`

```suggestion
                let file_in_app_dir = Path::new(app_dir)
                    .join(file.strip_prefix("./").unwrap_or_default())
                    .display()
                    .to_string();

                let file = if file.starts_with("./") {
                    &file_in_app_dir
                } else {
                    file
                };
```

A similar approach can be used in the above `get_copy_commands`